torch/testing/_comparison.py

```diff
@@ -0,0 +1,986 @@
+import abc
+import cmath
+import collections.abc
+from typing import NamedTuple, Callable, Sequence, List, Union, Optional, Type, Tuple, Any, cast
+
+import numpy as np
+
+import torch
+
+from ._core import _unravel_index
+
+
+class ErrorMeta(NamedTuple):
+    type: Type[Exception]
+    msg: str
+    id: Tuple[Any, ...] = ()
+
+    def to_error(self) -> Exception:
+        msg = self.msg
+        if self.id:
+            msg += f"\n\nThe failure occurred for item {''.join(str([item]) for item in self.id)}"
```

Comment:
"item" -> "item(s)"?

Reply:
Depends on what you mean. In `id` we store the nested indices / keys of the error. This does not multiple items, but rather a nested one. Example:

```python
>>> actual = dict(foo=[0, dict(bar=1, baz=[2])])
>>> expected = dict(foo=[0, dict(bar=1, baz=[3])])
>>> torch.testing.assert_close(actual, expected)
[...]
The failure occurred for item ['foo'][1]['baz'][0]
```

As long as we are talking about builtin types as keys one could now use the message to extract the values that are causing the error:
```python
>>> actual['foo'][1]['baz'][0]
2
>>> expected['foo'][1]['baz'][0]
3
```

Reply:
Ah, thanks for the clarification